frontend/app/src/features/questions/components/QuestionsUpsertModal.tsx

```diff
@@ -0,0 +1,106 @@
+import { Question } from "@/types/question";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Input,
+  FormControl,
+  FormLabel,
+  HStack,
+  VStack,
+} from "@chakra-ui/react";
+import { QUESTION_KEY_LABEL_MAP } from "../constants/questionKeyLabelMap";
+import { useQuestions } from "./QuestionsOutlet";
+import { useUpdateQuestion } from "../api/useUpdateQuestion";
+import { useCreateQuestion } from "../api/useCreateQuestion";
+
+const QuestionsUpsertModal = () => {
+  const {
+    isUpsertModalOpen,
+    onUpsertModalClose,
+    currQuestion,
+    setCurrQuestion,
+  } = useQuestions();
+  const updateQuestion = useUpdateQuestion();
+  const createQuestion = useCreateQuestion();
+
+  const handleSave = (question: Partial<Question> | undefined) => {
+    if (question) {
+      if (currQuestion) {
```

Comment:
This check does not work as intended:
- When creating a question, currQuestion is undefined
- But when I edit a field it updates currQuestion to a value so it ends up sending a PUT request instead of a POST request for creating a question

Reply:
Fixed in e7b20488a2a5767f4d52a6a64e605774eea1ebd1